pkg/operation/botanist/dnsrecord.go

```diff
@@ -206,3 +211,38 @@ func (b *Botanist) deployOrRestoreDNSRecord(ctx context.Context, dnsRecord compo
 	}
 	return dnsRecord.Deploy(ctx)
 }
+
+// CleanupOrphanedDNSRecordSecrets cleans up secrets related to DNSRecords which may be orphaned after introducing the 'dnsrecord-' prefix
+func (b *Botanist) CleanupOrphanedDNSRecordSecrets(ctx context.Context) error {
```

Comment:
I don't see this method used anywhere? Did you intend to use it in the flow?

Reply:
Indeed, this change was incomplete when I pushed it. I just wanted to put my current state of work out there for the discussion on how to implement this.